splitio/storage/adapters/redis.py

```diff
@@ -885,6 +886,74 @@ def _build_sentinel_client(config):  # pylint: disable=too-many-locals
     redis = sentinel.master_for(master_service)
     return RedisAdapter(redis, prefix=prefix)
 
+async def _build_sentinel_client_async(config):  # pylint: disable=too-many-locals
+    """
+    Build a redis client with sentinel replication.
+
+    :param config: Redis configuration properties.
+    :type config: dict
+
+    :return: A Wrapped redis-sentinel client
+    :rtype: splitio.storage.adapters.redis.RedisAdapter
+    """
+    sentinels = config.get('redisSentinels')
+
+    if config.get('redisSsl', False):
+        raise SentinelConfigurationException('Redis Sentinel cannot be used with SSL/TLS.')
+
+    if sentinels is None:
+        raise SentinelConfigurationException('redisSentinels must be specified.')
+    if not isinstance(sentinels, list):
+        raise SentinelConfigurationException('Sentinels must be an array of elements in the form of'
+                                             ' [(ip, port)].')
+    if not sentinels:
+        raise SentinelConfigurationException('It must be at least one sentinel.')
+    if not all(isinstance(s, tuple) for s in sentinels):
+        raise SentinelConfigurationException('Sentinels must respect the tuple structure'
+                                             '[(ip, port)].')
+
+    master_service = config.get('redisMasterService')
+
+    if master_service is None:
+        raise SentinelConfigurationException('redisMasterService must be specified.')
+
+    database = config.get('redisDb', 0)
+    password = config.get('redisPassword', None)
+    socket_timeout = config.get('redisSocketTimeout', None)
+    socket_connect_timeout = config.get('redisSocketConnectTimeout', None)
+    socket_keepalive = config.get('redisSocketKeepalive', None)
+    socket_keepalive_options = config.get('redisSocketKeepaliveOptions', None)
+    connection_pool = config.get('redisConnectionPool', None)
+    encoding = config.get('redisEncoding', 'utf-8')
+    encoding_errors = config.get('redisEncodingErrors', 'strict')
+    decode_responses = config.get('redisDecodeResponses', True)
+    retry_on_timeout = config.get('redisRetryOnTimeout', False)
+    max_connections = config.get('redisMaxConnections', None)
+    ssl = config.get('redisSsl', False)
```

Comment:
is this line necessary? (line 902 throws an error if `redisSsl` is provided)

Reply:
Ah, I need to remove the exception, since async redis sentinel does support ssl